pkg/services/ngalert/notifier/alert_reception.go

```diff
@@ -29,14 +34,15 @@ func NewAlertProvider(m types.Marker) (*AlertProvider, error) {
 }
 
 func (ap *AlertProvider) PutPostableAlert(postableAlerts apimodels.PostableAlerts) error {
```

Comment:
I have two questions, mostly for discussion.

1. Now that we're certain we don't need to do any custom logic here, do we even need a custom type? I'd be in favour of just deleting this file and using the mem alerts directly in the Alertmanager. 
2.  With that, we can move the current `endDate` logic and also include the call to `alert.Validate` in the method that receives the alerts. Think of it as an API handler but with an in-code call instead of HTTP.

Reply:
Right, I don't think this new `AlertProvider` is needed anymore. But I do think we need to accept the `apimodels.PostableAlerts` in Alertmanager, because if it comes to splitting this component, then handling the API will be trivial. The `Alertmanager.PutAlerts` can take `apimodels.PostableAlerts`, do the conversion, and send the usual memory alerts to the Prometheus Alertmanager alert provider.